draft-ietf-httpbis-connect-tcp.md

```diff
@@ -145,13 +145,13 @@ Authentication to a templated TCP proxy normally uses ordinary HTTP authenticati
 
 Clients SHOULD assume that all proxy resources generated by a single template share a protection space (i.e., a realm) ({{?RFC9110, Section 11.5}}).  For many authentication schemes, this will allow the client to avoid waiting for a "401 (Unauthorized)" response before each new connection through the proxy.
 
-### Use of the Capsule Protocol {#capsule-protocol}
+## Use of the Capsule Protocol {#capsule-protocol}
 
-In this specification, support for the Capsule Protocol {{?RFC9297}} is OPTIONAL for clients.  Clients MAY request use of the Capsule Protocol by including a "Capsule-Protocol: ?1" header field in the request.
+When using the "connect-tcp" Upgrade Token, templated TCP proxies do not use the Capsule Protocol {{!RFC9297}}.  Clients MAY request use of the Capsule Protocol by offering the Upgrade Token "connect-tcp-capsule" instead.  When offering or accepting the "connect-tcp-capsule" Upgrade Token, clients and servers SHOULD including a "Capsule-Protocol: ?1" header to facilitate processing by intermediaries.
 
-Server support for the Capsule Protocol is REQUIRED.  Servers MUST confirm use of the Capsule Protocol by echoing the "Capsule-Protocol: ?1" header in their response.
+Clients of this specification MAY implement "connect-tcp", "connect-tcp-capsule", or both.  Accordingly, templated TCP proxy servers for arbitrary clients MUST implement both Upgrade Tokens.
```

Comment:
I don't see why servers MUST implement both. One solution could be to use SHOULD instead (see below). Another option would be to add a sentence saying that servers that only expect one kind of client MAY only implement one of these Upgrade Tokens.
```suggestion
Clients of this specification MAY implement "connect-tcp", "connect-tcp-capsule", or both.  Accordingly, templated TCP proxy servers for arbitrary clients SHOULD implement both Upgrade Tokens to improve compatibility.
```

Reply:
OK, I've rephrased this to highlight an exemption for restricted deployments.